### PR TITLE
fix(backend): use random seed for SDE & Ancestral schedulers

### PR DESCRIPTION
**Pull request by psychedelicious**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes https://discord.com/channels/1020123559063990373/1133929466763161650/1137626552222359572
- [ ] No, because:

## Description

SDE and Ancestral schedulers use some randomness at each step when applying conditioning. We were using a static seed of `0` for this, regardless of the initial noise used. This could cause results to be a bit same-y.

Unfortunately, we do not have easy access to the seed used to create the initial noise at this time.

Changing this to use a random seed value instead of always 0.

@keturn

### CODE REVIEW

File: invokeai/app/invocations/latent.py [hunk 1]
```diff
@@ -198,7 +199,7 @@ def get_conditioning_data(
             # for ddim scheduler
             eta=0.0,  # ddim_eta
             # for ancestral and sde schedulers
-            generator=torch.Generator(device=unet.device).manual_seed(0),
+            generator=torch.Generator(device=unet.device).manual_seed(get_random_seed()),
```
Comment (keturn) [inline invokeai/app/invocations/latent.py:202]: If you're not using a specific seed, I don't think you need to `manual_seed` at all. But I guess it doesn't hurt.

**Review by keturn — APPROVED** [review]

File: invokeai/app/invocations/latent.py (hunk 1 above)
Comment (psychedelicious) [reply invokeai/app/invocations/latent.py:202]: Appears that the initial seed used is the same every time:
```py
>>> import torch
>>> torch.Generator().initial_seed()
67280421310721
# restart the repl just to be extra sure its fresh
>>> import torch
>>> torch.Generator().initial_seed()
67280421310721
```

But I did notice this function in the docs, which appears to be the blessed way to seed the generator:
```py
>>> g = torch.Generator()
>>> g.initial_seed()
67280421310721
>>> g.seed()
15390967034040818360
>>> g.initial_seed()
15390967034040818360
>>> g.seed()
11307471257349742134
>>> g.seed()
17364437299175068371
>>> g.seed()
5841587831953547992
```

I'm changing the PR to use that.

**Comment by psychedelicious** [comment]
This is superseded by #4201 in which @StAlKeR7779 has added a `seed` prop to `LatentsField` to provide context to downstream nodes. The SDE and Ancestral samplers can then use this to provide deterministic outputs.
